### PR TITLE
Disable dynamic mapping

### PR DESCRIPTION
**Pull request by VRuzhentsov**
because default toSearchableArray() method filling up with new mappings

### CODE REVIEW

**Comment by thomasjsn** [comment]
I don't see dynamic mapping as being a problem. Just create the mappings first, before importing. If disabling them makes sense then it has to go in the config file, not hard coded in the command.

**Comment by VRuzhentsov** [comment]
about config, its a good idea.

When i creating mappings first, via `:make-indices`, in the beginning, it is ok, but when I importing a couple of models with `searchable()` method, all data are going to `[body][doc]`, and ES see, that it is new data, and create a new mapping based on `toSearchableArray()` method, which by default is calling` Model->toArray()`. The last one could catch model relations, in case if some variable like $with is written, and a lot of superfluous mapping

**Comment by VRuzhentsov** [comment]
And, about  ErickTamayo request of single config file for scout. Is it ok, that i move all yours config to scout config?

**Comment by thomasjsn** [comment]
I don't understand. If you create the mapping first, and then import data. This will not make ES create dynamic mappings, the mappings already exist. If you import fields that you have not created mapping for, then yes; dynamic mapping will be used. To test this, create a field mapping and import data for that field which is inconsistent with mapping, you'll get an error.

I override `toSearchableArray()` in all my models. Because I want to tailor what gets indexed. If it makes sense to do so I also index relationships, by explicitly defining them in `toSearchableArray()`. Just remember that by default the model observer will not update a document if a relationship changes. meaning that the document data could be outdated.

I don't agree that the config should be in the Scout file, it is Elastic specific and should be in a separate config file. But feel free to fork my fork and change it there :+1:

**Comment by thomasjsn** [comment]
And by "But feel free to fork my fork and change it there" I mean make a pull request in ErickTamayo repo.

**Comment by VRuzhentsov** [comment]
did you mean, about this guy, with empty commits?

**Comment by thomasjsn** [comment]
No I just reread my last comment and saw that maybe my intentions where not as clear as I thought :)

**Comment by VRuzhentsov** [comment]
About several indexes. As I see in other realisations of ES in laravel, they are using single index for single application. Why did you decide to do configurable number of indexes?

File: src/Console/ElasticMakeIndicesCommand.php [hunk 1]
```diff
@@ -47,7 +47,7 @@ public function handle()
 
         foreach ($indices as $index) {
 
-            $indexConfig = config("elasticsearch.indices.{$index}");
+            $indexConfig = config("elasticsearch.indices.$index");
```
Comment (VRuzhentsov) [inline src/Console/ElasticMakeIndicesCommand.php:50]: I  can rollback, in reason of not too strong in PHP

**Comment by VRuzhentsov** [comment]
And about contributing to ErickTamayo repo, you want in such way to motivate Eric to accept your idea?  Because I agree with you, and my changes are based on yours

**Comment by thomasjsn** [comment]
I have different document types that have incompatible mappings. That doesn't work if they are in the same index. Further I like not having to reindex everything if I change mappings, but one the section affected.

I have to admit; I still don't see why you have to disable dynamic mapping... Your defined mapping will not be overwritten by something dynamic. Dynamic mapping is only created if you didn't do it manually.

Well, I'm not really concerned with if he accepts it or not. I think we want different things for the package. I was just answering your question:

> And, about ErickTamayo request of single config file for scout. Is it ok, that i move all yours config to scout config?

I'd like it to be a separate config file, as it is now.

**Comment by thomasjsn** [comment]
I suggest you do exactly what you need in your own repo. I don't really have the time to review/test. I'm sorry, I just need it to work for my project.

**Pull request closed without merging**